applications/IgaApplication/custom_elements/embedded_isogeometric_beam_element.cpp

```diff
@@ -0,0 +1,5629 @@
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_X));
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_Y));
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_Z));
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(ROTATION_X));
+        }
+        KRATOS_CATCH("")
+    };
+
+    ///@}
+    ///@name Analysis stages
+    ///@{
+
+    void EmbeddedIsogeometricBeamElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+        KRATOS_CATCH("")
+        //InitializeMaterial();
+    }
+
+    void EmbeddedIsogeometricBeamElement::InitializeMaterial()
+    {
+        KRATOS_TRY
+
+        const GeometryType& r_geometry = GetGeometry();
+        const Properties& r_properties = GetProperties();
+        const auto& r_N = r_geometry.ShapeFunctionsValues();
+
+        const SizeType r_number_of_integration_points = r_geometry.IntegrationPointsNumber();
+
+        //Constitutive Law initialisation
+        if (mConstitutiveLawVector.size() != r_number_of_integration_points)
+            mConstitutiveLawVector.resize(r_number_of_integration_points);
+
+        for (IndexType point_number = 0; point_number < r_number_of_integration_points; ++point_number) {
+            mConstitutiveLawVector[point_number] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
+            mConstitutiveLawVector[point_number]->InitializeMaterial(r_properties, r_geometry, row(r_N, point_number));
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    void EmbeddedIsogeometricBeamElement::CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo,
+        const bool ComputeLeftHandSide,
+        const bool ComputeRightHandSide)
+    {
+        //KRATOS_WATCH("EmbeddedIsogeometricBeamElement::CalculateAll");
+        //KRATOS_WATCH(this->Id());
+        set_Memory();
+        const auto& r_geometry = GetGeometry();
+        auto& r_integration_points = r_geometry.IntegrationPoints();
+            
+        int point_number = 0;
+        float _dL;
+        //// get integration data and weights, GP weight and dL and (u_vec[Knotspan_index+1]-u_vec[Knotspan_index])/2
+        const double& integration_weight = r_integration_points[point_number].Weight();
+
+        _gke.clear();
+        _gfie.clear();
+        //stiff_mat_el_lin(rCurrentProcessInfo, point_number, _gke, _gfie, _dL);
+        stiff_mat_el_nln(rCurrentProcessInfo, point_number, _gke, _gfie, _dL);
+        //stiff_mat_el_geo(rCurrentProcessInfo, point_number, _gke,  _dL);
+        
+        float mult = (integration_weight * _dL);
+ 
+
+        if (ComputeLeftHandSide == true)
+        {
+            //KRATOS_WATCH(_gke);
+            //KRATOS_WATCH(rLeftHandSideMatrix);
+            //KRATOS_WATCH(r_integration_points[point_number].Coordinates()[0]);
+            noalias(rLeftHandSideMatrix) += mult * _gke;
```

Comment:
besides, these names are not really meaningful, makes super difficult to follow the methodology